biodatasets/meddialog/meddialog.py

```diff
@@ -0,0 +1,216 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import re
+import datasets
+from pathlib import Path
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+
+_DATASETNAME = "meddialog"
+
+_CITATION = """
+@article{DBLP:journals/corr/abs-2004-03329,
+  author    = {Shu Chen and
+               Zeqian Ju and
+               Xiangyu Dong and
+               Hongchao Fang and
+               Sicheng Wang and
+               Yue Yang and
+               Jiaqi Zeng and
+               Ruisi Zhang and
+               Ruoyu Zhang and
+               Meng Zhou and
+               Penghui Zhu and
+               Pengtao Xie},
+  title     = {MedDialog: {A} Large-scale Medical Dialogue Dataset},
+  journal   = {CoRR},
+  volume    = {abs/2004.03329},
+  year      = {2020},
+  url       = {https://arxiv.org/abs/2004.03329},
+  eprinttype = {arXiv},
+  eprint    = {2004.03329},
+  biburl    = {https://dblp.org/rec/journals/corr/abs-2004-03329.bib},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+"""
+
+_DESCRIPTION = """
+The MedDialog dataset (English) contains conversations (in English) between doctors and patients.\
+It has 0.26 million dialogues. The data is continuously growing and more dialogues will be added. \
+The raw dialogues are from healthcaremagic.com and icliniq.com.\
+
+All copyrights of the data belong to healthcaremagic.com and icliniq.com.
+"""
+
+_HOMEPAGE = "https://github.com/UCSD-AI4H/Medical-Dialogue-System"
+
+_LICENSE = "Public for Research"
+
+_URLs = {
+    "en": {
+        "train": "https://drive.google.com/file/d/1ria4E6IdTIPsikL4Glm3uy1tFKJKw0W8/view?usp=sharing",
+        "validation": "https://drive.google.com/file/d/1KAZneuwdfEVQQM6euCX4pMDP-9DQpiB5/view?usp=sharing",
+        "test": "https://drive.google.com/file/d/10izqL71kcgnteYsf87Vh6j_mZ8sZM2Rc/view?usp=sharing",
+    },
+    "zh": {
+        "train": "https://drive.google.com/file/d/1AaDJoHaiHAwEZwtskRH8oL1UP4FRgmgx/view?usp=sharing",
+        "validation": "https://drive.google.com/file/d/1TvfZCmQqP1kURIfEinOcj5VOPelTuGwI/view?usp=sharing",
+        "test": "https://drive.google.com/file/d/1pmmG95Yl6mMXRXDDSRb9-bYTxOE7ank5/view?usp=sharing",
+    },
+}
+
+_SUPPORTED_TASKS = [Tasks.TEXT_CLASSIFICATION]
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+class MedDialog(datasets.GeneratorBasedBuilder):
+    """MedDialog: Large-scale Medical Dialogue Datasets in English and Chinese."""
+
+    DEFAULT_CONFIG_NAME = "meddialog_en_source"
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        # Source schemas
+        BigBioConfig(
+            name="meddialog_en_source",
+            version=SOURCE_VERSION,
+            description="MedDialog source schema",
+            schema="source",
+            subset_id="meddialog",
```

Comment:
I think we should make each language a subset id 
```suggestion
            subset_id="meddialog_en",
```